src/components/sidebar/routePathView/LinkListView.tsx

```diff
@@ -0,0 +1,195 @@
+    BORDER: 'Raja',
+    ALL: 'Näytä kaikki',
+};
+
+@inject('sidebarStore')
+@observer
+class ILinkListView extends React.Component<ILinkListViewProps, ILinkListViewState>{
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            selectedRoutePathLink: undefined,
+            linkTableFilter:  NodeType.STOP,
+        };
+    }
+
+    public selectRoutePathLink = (id: string) =>
+    (event: React.MouseEvent<HTMLElement>) => {
+        this.setState({
+            selectedRoutePathLink: id,
+        });
+    }
+
+    private getRoutePathLinks = () => {
+        const routePathLinks = this.props.routePath.routePathLinks;
+        if (!routePathLinks) return;
+        return routePathLinks.map((routePathLink) => {
+            if (routePathLink.startNodeType === this.state.linkTableFilter ||
+                this.state.linkTableFilter === filterTypes.ALL) {
+                return(
+                    <div
+                        key={routePathLink.id}
+                        className={
+                            (this.state.selectedRoutePathLink === routePathLink.id)
+                            ? s.routePathLinkRowSelected : s.routePathLinkRow}
+                        onClick={this.selectRoutePathLink(routePathLink.id)}
+                    >
+                        <div className={s.flexInnerRow}>
+                            <div className={s.tableItemMedium}>
+                                {routePathLink.id}
+                            </div>
+
+                            <div className={s.tableItem}>
+                                {this.getNodeName(routePathLink.startNode)}
+                            </div>
+
+                            <div className={s.tableItem}>
+                                {this.getNodeName(routePathLink.endNode)}
+                            </div>
+
+                            <div className={s.tableItem}>
+                                {this.getHastusId(routePathLink)}
+                            </div>
+
+                            <div className={s.tableItemSmall}>
+                                {routePathLink.startNodeType}
+                            </div>
+                        </div>
+                    </div>);
+            }
+            return;
+        });
+    }
+
+    public getHastusId = (routePathLink: IRoutePathLink) => {
+        return (routePathLink.startNode.stop && routePathLink.startNode.stop.hastusId) ?
+            routePathLink.startNode.stop.hastusId :
+            '-';
+    }
+
+    public getNodeName = (node: INode) => {
+        return (node.stop) ?
+            node.stop.nameFi :
+            '-';
+    }
+
+    public openLinkView = () => {
+        if (this.state.selectedRoutePathLink) {
+            this.props.sidebarStore!.setOpenLinkId(parseInt(this.state.selectedRoutePathLink, 10));
+            const linkViewLink = routeBuilder.to(subSites.link).toLink();
+            window.open(linkViewLink);
```

Comment:
One reason why we are using react router instead of normal links, is that in normal links (such as window.open) will fetch the whole SPA again. React router links wont trigger a page update, and wont load the whole SPA, instead it just trigger react changes to view a new page.
Navigator.goTo(linkViewLink) is therefore faster.

Reply:
I originally used Navigator.goTo(linkViewLink) but figured it would be more user friendly to open the link page in a new window.

Reply:
Oh, I see.